powersimdata/design/investment/investment_costs.py

```diff
@@ -381,7 +382,7 @@ def calculate_gen_inv_costs(
     grid = scenario.state.get_grid()
     if base_grid is None:
         base_grid = Grid(
-            scenario.info["interconnect"].split("_"), source=scenario.info["grid_model"]
+            model_data=AbstractGridFactory.get_or_create(scenario.info["grid_model"], scenario.info["interconnect"].split("_"))
```

Comment:
I still think it wouldn't hurt to have a `get_base_grid` method in the scenario class to simplify usages like this.

Reply:
Unrelated - looks like we still need to handle the case where we pass a string for the grid model, right? In `get_or_create`, we call `__init__` directly on the argument. Same comment for passing a mat file path.

Reply:
>Unrelated - looks like we still need to handle the case where we pass a string for the grid model, right? In get_or_create, we call __init__ directly on the argument. Same comment for passing a mat file path.

@jon-hagg In the proposed changes, it's no longer possible to pass a string for the grid model. Specifically, this is why `set_grid` was changed to take in an `AbstractGrid` object instead of a string (you'll notice the default value for `grid_model` is now `TAMU` instead of `"usa_tamu"`). The `scenario.info["grid_model"]` will then always be an `AbstractGrid` object